llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -456,15 +537,23 @@ int main(int argc, char **argv) {
     Error Err = Table.addColumn(COL_CODE, Files);
     CHECK_AND_EXIT(Err);
   }
-  if (DoSpecConst && SetSpecConstAtRT) {
-    // extract spec constant maps per each module
-    for (auto &MUptr : ResultModules) {
-      ResultSpecIDMaps.emplace_back(SpecIDMapTy());
-      if (SpecConstsMet)
-        SpecConstantsPass::collectSpecConstantMetadata(*MUptr.get(),
-                                                       ResultSpecIDMaps.back());
+  {
+    // Device library req mask is collected and stored in device image property
+    // as default and each device image module will have one req mask.
+    std::vector<uint32_t> DeviceLibReqMaskVec;
+    getDeviceLibReqMasks(ResultModules, DeviceLibReqMaskVec);
+    if (DoSpecConst && SetSpecConstAtRT) {
+      // extract spec constant maps per each module
+      for (auto &MUptr : ResultModules) {
+        ResultSpecIDMaps.emplace_back(SpecIDMapTy());
+        if (SpecConstsMet)
+          SpecConstantsPass::collectSpecConstantMetadata(
+              *MUptr.get(), ResultSpecIDMaps.back());
+      }
+      assert(DeviceLibReqMaskVec.size() == ResultSpecIDMaps.size());
```

Comment:
Why this assertion is needed? It will crash if `(DoSpecConst && SetSpecConstAtRT) == false`, right?

Reply:
For each sub module, we will have one "require mask" stored in property file. And if (DoSpecConst && SetConstAtRt) == true, we will store one SpecIDMap for each sub module, so the number of "require mask" and "SpecIDMaps" should be the same.
If (DoSpecConst && SetSpecConstAtRT) == false, the assert will be skipped and we will not store SpecIDMaps in prop file for each submodule.

Reply:
You're right, I got the indentation wrong.